clients/lsp-terraform.el

```diff
@@ -222,32 +228,55 @@ This is a synchronous action."
 (declare-function lsp-treemacs-render "ext:lsp-treemacs" (tree title expand-depth &optional buffer-name))
 
 (defvar-local lsp-terraform-ls--providers-tree-data nil)
+(defvar-local lsp-terraform-ls--modules-call-tree-data nil)
 (defconst lsp-terraform-ls--providers-buffer-name "*Terraform Providers*")
+(defconst lsp-terraform-ls--modules-buffer-name "*Terraform Modules*")
 
 (cl-defstruct tf-package display-name doc-link installed-version version-constraint)
 
+(cl-defstruct tf-module name doc-link version source-type dependent-modules)
+
 (defun construct-tf-package (provider installed-version)
   "Construct `TF-PACKAGE' using PROVIDER and INSTALLED-VERSION."
-  (make-tf-package :display-name (gethash "display_name" provider)
-                   :doc-link (gethash "docs_link" provider)
+  (make-tf-package :display-name (lsp-get provider :display_name)
```

Comment:
typically here you can use the generated getters/setters or even better use destructoring by utilizing lsp-defun.

Reply:
I did consider that, but the field `installed-version` comes from a different structure.

But I will keep that in mind when writing future code, thanks!